scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,824 @@
+        try {
+            if (persistedChannels.containsKey(name)) {
+                persistedChannels.remove(name);
+                logWithContext(taskid, null, "Deleted persistent channel \'" + name + "\'", Level.INFO);
+                recMan.commit();
+                return true;
+            } else if (inMemoryChannels.containsKey(name)) {
+                inMemoryChannels.remove(name);
+                logWithContext(taskid, null, "Deleted memory channel \'" + name + "\'", Level.INFO);
+                return true;
+            } else {
+                return false;
+            }
+        } catch (InvalidChannelException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Override
+    public <T extends Serializable> boolean createChannelIfAbsent(String originator, TaskId taskid, String name,
+            boolean isPersistent) throws IOException {
+        if (inMemoryChannels.containsKey(name) || persistedChannels.containsKey(name)) {
+            return false;
+        } else {
+            createChannel(originator, taskid, name, isPersistent);
+            return true;
+        }
+    }
+
+    private void commitIfNeeded(String channel) throws IOException {
+        if (persistedChannels.containsKey(channel)) {
+            persistedChannels.put(channel, persistedChannels.get(channel));
+            recMan.commit();
+        }
+    }
+
+    @Override
+    public boolean containsKey(String originator, TaskId taskid, String channel, String key)
+            throws InvalidChannelException {
+        return getChannel(channel).containsKey(key);
+    }
+
+    @Override
+    public int size(String originator, TaskId taskid, String channel) throws InvalidChannelException {
+        return getChannel(channel).size();
+    }
+
+    @Override
+    public boolean isEmpty(String originator, TaskId taskid, String channel) throws InvalidChannelException {
+        return getChannel(channel).isEmpty();
+    }
+
+    @Override
+    public boolean containsValue(String originator, TaskId taskid, String channel, Serializable value)
+            throws InvalidChannelException {
+        return getChannel(channel).containsValue(value);
+    }
+
+    @Override
+    public <T extends Serializable> T get(String originator, TaskId taskid, String channel, String key)
+            throws InvalidChannelException {
+        return ((Channel<T>) getChannel(channel)).get(key);
+    }
+
+    @Override
+    public <T extends Serializable> T put(String originator, TaskId taskid, String channel, String key, T value)
+            throws InvalidChannelException, IOException {
+        T answer = ((Channel<T>) getChannel(channel)).put(key, value);
+        commitIfNeeded(channel);
+        logWithContext(taskid,
+                       channel,
+                       "Put " + value + " on key '" + key + "', previous value was " + answer + "",
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "', previous value was " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)